packages/multichain-network-controller/src/MultichainNetworkService/MultichainNetworkService.ts

```diff
@@ -15,19 +15,58 @@ import {
 export class MultichainNetworkService {
   readonly #fetch: typeof fetch;
 
+  private readonly BATCH_SIZE = 20;
+
   constructor({ fetch: fetchFunction }: { fetch: typeof fetch }) {
     this.#fetch = fetchFunction;
   }
 
   /**
    * Fetches active networks for the given account IDs.
+   * Automatically handles batching requests to comply with URL length limitations.
    *
    * @param accountIds - Array of CAIP-10 account IDs to fetch activity for.
-   * @returns Promise resolving to the active networks response.
+   * @returns Promise resolving to the combined active networks response.
    * @throws Error if the response format is invalid or the request fails.
    */
   async fetchNetworkActivity(
     accountIds: CaipAccountId[],
+  ): Promise<ActiveNetworksResponse> {
+    if (accountIds.length === 0) {
+      return { activeNetworks: [] };
+    }
+
+    if (accountIds.length <= this.BATCH_SIZE) {
+      return this.#fetchNetworkActivityBatch(accountIds);
+    }
+
+    const batches: CaipAccountId[][] = [];
+    for (let i = 0; i < accountIds.length; i += this.BATCH_SIZE) {
+      batches.push(accountIds.slice(i, i + this.BATCH_SIZE));
+    }
```

Comment:
Very nit: But we could use `lodash.chunk` here`:

```suggestion
    const batches = chunk(accountIds, this.BATCH_SIZE);
```

Reply:
Great suggestion

Reply:
Added here [3abb0b0](https://github.com/MetaMask/core/pull/5752/commits/3abb0b0e2b0a41cf0a0b28e54e43094af199a15a)